DataFormats/Math/interface/GraphUtil.h

```diff
@@ -0,0 +1,76 @@
+#ifndef DATA_FORMATS_MATH_GRAPH_UTIL_H
+#define DATA_FORMATS_MATH_GRAPH_UTIL_H
+
+#include "DataFormats/Math/interface/Graph.h"
+#include "DataFormats/Math/interface/GraphWalker.h"
+#include <iostream>
+#include <string>
+
+template<class N, class E>
+  void output(const math::Graph<N,E> & g, const N & root)
+{
+  math::GraphWalker<N,E> w(g,root);
+  bool go=true;
+  while(go) {
+    std::cout << w.current().first << ' ';
+    go=w.next();
+  }
+  std::cout << std::endl;
+}
+
+template<class N, class E>
+  void graph_combine(const math::Graph<N,E> & g1, const math::Graph<N,E> & g2,
+		     const N & n1, const N & n2, const N & root,
+		     math::Graph<N,E> & result)
+{
+  result = g1;
+  result.replace(n1,n2);
+  //output(result,n2);
+  math::GraphWalker<N,E> walker(g2,n2);
+  while (walker.next()) {
+    const N & parent = g2.nodeData((++walker.stack().rbegin())->first->first);
+    /*
+      N parent = g2.nodeData((++walker.stack().rbegin())->first->first);
+      N child  = walker.current().first;
+      E edge   = walker.current().second;
+    */
+    //std::cout << parent << ' ' << walker.current().first << ' ' << walker.current().second<< std::endl;
+    result.addEdge(parent, walker.current().first, walker.current().second);
+    //result.addEdge(parent,child,edge);
+    //result.dump_graph();
+    //output(result,n2);
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant